maploc/data/dataset.py

```diff
@@ -126,44 +127,74 @@ def get_view(self, idx, scene, seq, name, seed, bbox_tile):
         elif "camera_height" in self.data:
             data["camera_height"] = self.data["height"][idx].clone()
 
         # raster extraction
         canvas = self.tile_managers[scene].query(bbox_tile)
-        xy_w_gt = self.data["t_c2w"][idx][:2].numpy()
-        uv_gt = canvas.to_uv(xy_w_gt)
-        uv_init = canvas.to_uv(bbox_tile.center)
         raster = canvas.raster  # C, H, W
 
+        world_T_cam = Transform3D.from_Rt(
+            self.data["R_c2w"][idx], self.data["t_c2w"][idx]
+        ).float()
+        world_T_cam2d = Transform2D.from_Transform3D(
+            world_T_cam
+        )  # angle is b/w world x and cam z
+
+        # gcam: gravity-aligned camera with z=optical axis
+        # gcamxyz: gcam rotated such that z:up,x:right,y:forward.
+
+        gcam_angle = world_T_cam2d.angle - 90
+        Rz = Transform2D.from_degrees(gcam_angle, torch.zeros(2))
+        world_R_gcamxyz = torch.eye(3)
+        world_R_gcamxyz[:2, :2] = Rz.R
+        world_T_gcamxyz = Transform3D.from_Rt(world_R_gcamxyz, world_T_cam.t).float()
+        gcamxyz_T_gcam = Transform3D.from_Rt(
+            Rotation.from_euler("X", -90, degrees=True).as_matrix(), torch.zeros(3)
+        ).float()
+        world_T_gcam = world_T_gcamxyz @ gcamxyz_T_gcam
+        gcam_T_cam = world_T_gcam.inv() @ world_T_cam
+
+        world_T_tile = Transform2D.from_Rt(torch.eye(2), canvas.bbox.min_).float()
+        tile_T_cam = world_T_tile.inv() @ world_T_cam2d
+        map_T_cam = Transform2D.from_Rt(
+            tile_T_cam.R, tile_T_cam.t * canvas.ppm - 0.5
+        )  # This will be deprecated, tile_T_cam is sufficient.
+
+        world_T_init = Transform2D.from_Rt(torch.eye(2), bbox_tile.center).float()
+        tile_T_init = world_T_tile.inv() @ world_T_init
+        map_T_init = Transform2D.from_Rt(torch.eye(2), tile_T_init.t * canvas.ppm - 0.5)
+
+        yaw = tile_T_cam.angle
+
         # Map augmentations
-        heading = np.deg2rad(90 - yaw)  # fixme
-        if self.stage == "train":
-            if self.cfg.augmentation.rot90:
-                raster, uv_gt, heading = random_rot90(raster, uv_gt, heading, seed)
-            if self.cfg.augmentation.flip:
-                image, raster, uv_gt, heading = random_flip(
-                    image, raster, uv_gt, heading, seed
-                )
-        yaw = 90 - np.rad2deg(heading)  # fixme
-
-        image, valid, cam, roll, pitch = self.process_image(
-            image, cam, roll, pitch, seed
-        )
+        # TODO: refactor
+        # heading = np.deg2rad(90 - yaw)  # fixme
+        # if self.stage == "train":
+        #     if self.cfg.augmentation.rot90:
+        #         raster, uv_gt, heading = random_rot90(raster, uv_gt, heading, seed)
+        #     if self.cfg.augmentation.flip:
+        #         image, raster, uv_gt, heading = random_flip(
+        #             image, raster, uv_gt, heading, seed
+        #         )
+        # yaw = 90 - np.rad2deg(heading)  # fixme
```

Comment:
Will this be address later in this PR?

Reply:
Yes, pushed [here](https://github.com/facebookresearch/OrienterNet/pull/49/commits/589de6f916128ed03a909d1adce50eb27541c154).

This should also fix #50, as the rectification is done before the image is flipped. However, if we want to allow the model to use cam_R_gcam in the future, we will need to update cam_R_gcam and add it to the dataloader's output. I can add that to this PR if needed.